src/FxCop/System.Runtime.Analyzers/CSharp/Design/TypesThatOwnDisposableFieldsShouldBeDisposable.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace System.Runtime.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class CSharpTypesThatOwnDisposableFieldsShouldBeDisposableAnalyzer : TypesThatOwnDisposableFieldsShouldBeDisposableAnalyzer<TypeDeclarationSyntax>
+    {
+        protected override DisposableFieldAnalyzer GetAnalyzer(INamedTypeSymbol disposableTypeSymbol)
+        {
+            return new CSharpDisposableFieldAnalyzer(disposableTypeSymbol);
+        }
+
+        protected sealed class CSharpDisposableFieldAnalyzer : DisposableFieldAnalyzer
+        {
+            public CSharpDisposableFieldAnalyzer(INamedTypeSymbol disposableTypeSymbol)
+                : base(disposableTypeSymbol)
+            { }
+
+            protected override bool IsDisposableFieldCreation(SyntaxNode node, SemanticModel model, HashSet<ISymbol> disposableFields, CancellationToken cancellationToken)
+            { 
+                if (node is AssignmentExpressionSyntax)
+                {
+                    var assignment = (AssignmentExpressionSyntax)node;
+                    if (assignment.Right is ObjectCreationExpressionSyntax &&
+                        disposableFields.Contains(model.GetSymbolInfo(assignment.Left, cancellationToken).Symbol))
+                    {
+                        return true;
+                    }    
+                }
+                else if (node is FieldDeclarationSyntax)
+                {
+                    var fieldDecl = ((FieldDeclarationSyntax)node).Declaration;
+                    Debug.Assert(fieldDecl.Variables.Count == 1);
+                    var fieldInit = fieldDecl.Variables.First();
+                    if (fieldInit?.Initializer?.Value is ObjectCreationExpressionSyntax &&
+                        disposableFields.Contains(model.GetDeclaredSymbol(fieldInit, cancellationToken)))
+                    {
```

Comment:
This code doesn't handle situations like this properly:

``` C#
IDisposable x, y = Foo(), z = Bar();
```

In debug builds the assert will fire because there are three `fieldDecl.Variables.Count` will be 3, not 1. Also, it will simply not examine `y` and `z`.

You need to consider all `VariableDeclaratorSyntax` nodes under `FieldDeclaration` nodes, not the `FieldDeclaration` nodes themselves.